src/control/lib/control/request.go

```diff
@@ -60,13 +60,20 @@ type (
 		getDeadline() time.Time
 	}
 
+	// sysSetter defines an interface to be implemented by
+	// requests that can set the system name Field.
+	sysSetter interface {
+		SetSystem(string)
```

Comment:
This interface should probably include `getSystem() string` as well, so that you can check to see if the system has already been set on the request.

Reply:
getSystem() is only ever called on a control API request which always embeds request stuct, getSystem() is implemented on the base request struct. is there a specific reason for providing getSystem() in the sysSetter interface that I'm not understanding?